watcher/lib/rollups.go

```diff
@@ -0,0 +1,145 @@
+package lib
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"strconv"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// Rollup is the rollup configuration
+type Rollup struct {
+	// Retention is the retention policy for this rollup
+	Retention string `json:"retention"`
+	// Measurement is the name of the measurements to run rollup on
+	Measurement string `json:"measurement"`
+	// Name is both the name of the rollup query and the name of the
+	// new measurement rollup data will be inserted into
+	Name string `json:"name"`
+	// Functions is a list of functions for rollup calculation
+	Functions []Function `json:"functions"`
+}
+
+// Check verifies that rollup configuration is correct
+func (r Rollup) Check() error {
+	if !OneOf(r.Retention, AllRetentions) {
```

Comment:
you should also make sure to keep `retentionToInterval` in sync with `AllRetentions` or have it build from the same source